extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -782,8 +796,17 @@ private static List<MethodInfo> getMethods(ClassInfo declaringClass, IndexView i
                 }
             }
         }
-        return methods;
 
+        DotName superClassName = declaringClass.superName();
+        if (superClassName != null) {
+            ClassInfo superClass = index.getClassByName(superClassName);
+
+            if (superClass != null) {
+                methods.addAll(getMethods(superClass, index));
+            }
+        }
+
+        return methods;
```

Comment:
Should we handle the case with a deeper hierarchy?

Reply:
Line 805 here recurses into the same method up until `java.lang.Object`. Granted, there is no test case for multi-level, but I can add that.

Reply:
I've added a multi-level resource class hierarchy to the existing test case.